framework/web/Request.php

```diff
@@ -187,6 +188,24 @@ public function resolve()
             }
             return [$route, $this->getQueryParams()];
         } else {
+            if (Yii::$app->getUrlManager()->enablePrettyUrl && !empty(Yii::$app->getUrlManager()->suffix)) {
+                $pathInfo = $this->getPathInfo();
+                $suffix = Yii::$app->getUrlManager()->suffix;
+                $n = strlen($suffix);
+                if (substr_compare($pathInfo, $suffix, -$n, $n) !== 0) {
+                    $this->setPathInfo($pathInfo . $suffix);
```

Comment:
you are changing the current Path to include the $suffix. When it still is not resolving, the NotFoundHttpException will have an updated Request and not the original. This would be confusing.